pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -435,6 +435,10 @@ class ClassTypeHintMatch:
 }
 # Overriding properties and functions are normally checked by mypy, and will only
 # be checked by pylint when --ignore-missing-annotations is False
+_BASE_DEVICE_CLASS = TypeHintMatch(
+    function_name="device_class",
+    return_type=["str", None],
+)
```

Comment:
Let's not overcomplicate things. I'm not sure it does make sense to extract individual `TypeHintMatches` only because they occur more than once.

--
Extracting all entity methods and properties, like you have done already, is something different.

Reply:
I didn't like it either, so I've adjusted it so that base entities accept all xxxxDeviceClass.

Reply:
Adjusted again to define device_class inside each platform and not inside the base entity